modules/__tests__/LocationUtils-test.js

```diff
@@ -160,3 +160,71 @@ describe("createLocation", () => {
     });
   });
 });
+
+describe("shouldReplace", () => {
+  it("should return true if current path is the same than new path", () => {
+    const currentLocation = {
+      pathname: "the/path",
+      search: "?the=query",
+      hash: "#the-hash"
+    };
+    expect(shouldReplace(currentLocation, 'the/path?the=query#the-hash')).toBeTruthy();
+  });
+
+  it("should return false if current path is not the same than new path", () => {
+    const currentLocation = {
+      pathname: "the/path",
+      search: "?the=query",
+      hash: "#the-hash"
+    };
+    expect(shouldReplace(currentLocation, 'the/path?the=query')).toBeFalsy();
+  });
+
+  it("should return true if current path is the same than new path and current state is the same than new state", () => {
+    const currentLocation = {
+      pathname: "the/path",
+      search: "?the=query",
+      hash: "#the-hash",
+      state: {
+        foo: 'bar',
+        obj: {
+          foo2: 'var2'
+        }
+      }
+    };
+    expect(shouldReplace(
+      currentLocation, 
+      'the/path?the=query#the-hash',
+      {
+        foo: 'bar',
+        obj: {
+          foo2: 'var2'
+        }
+      }
+    )).toBeTruthy();
+  });
+
+  it("should return false if current path is the same than new path and current state is not the same than new state", () => {
+    const currentLocation = {
+      pathname: "the/path",
+      search: "?the=query",
+      hash: "#the-hash",
+      state: {
+        foo: 'bar',
+        obj: {
+          foo2: 'var2'
+        }
+      }
+    };
+    expect(shouldReplace(
+      currentLocation, 
+      'the/path?the=query#the-hash',
+      {
+        foo: 'bar',
+        obj: {
+          foo2: 'var3'
+        }
+      }
+    )).toBeFalsy();
+  });
+});
```

Comment:
I think there's supposed to be a newline at the end of each file.

Reply:
Done, anyway linter does not have this rule. If you prefer it would be fine if it has it.